sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3752,11 +3780,18 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
   PI_ASSERT((WorkDim > 0) && (WorkDim < 4), PI_INVALID_WORK_DIMENSION);
 
   if (GlobalWorkOffset != NULL) {
-    for (pi_uint32 i = 0; i < WorkDim; i++) {
-      if (GlobalWorkOffset[i] != 0) {
-        return PI_INVALID_VALUE;
-      }
+    ze_driver_extension_properties_t GlobalOffsetExtension{
+        ZE_GLOBAL_OFFSET_EXP_NAME, ZE_GLOBAL_OFFSET_EXP_VERSION_1_0};
+
+    auto Platform = Queue->Device->Platform;
+    if (!(Platform->findDriverExtension(GlobalOffsetExtension))) {
```

Comment:
can we further update some local "static bool" value such that the map lookup is performed only once?

Reply:
sure, that can be done during the initialization. In that case, defining this generic helper function findDriverExtension() won't be necessary anymore in this PR